llama/runner/newrunner/cache.go

```diff
@@ -0,0 +1,251 @@
+package newrunner
```

Comment:
Do we want to put this under a root-level `runner` package, and leave the old one under `llama/runner`?

Reply:
My preference is to keep the two runners together since they have a similar structure and share some code where possible. I think the best thing to do would be to move the whole `llama/runner` directory with both old and runners into the root level.